### PR TITLE
CRM-15049: Fixed include/exclude search for parent groups

### PR DESCRIPTION
**Pull request by ahmedg1986**
I fixed issues as per below JIRA ticket:
https://issues.civicrm.org/jira/browse/CRM-15049

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins ok to test

**Comment by seamuslee001** [comment]
@ahmedg1986 can you please fix the style errors https://test.civicrm.org/job/CiviCRM-Core-PR/15916/checkstyleResult/new/

**Comment by ahmedg1986** [comment]
@seamuslee001 - I've fixed it

File: CRM/Contact/Form/Search/Custom/Group.php [hunk 1]
```diff
@@ -260,6 +260,16 @@ public function from() {
...
 
+      /*
+       * CRM-15049 - Bug Fix for Include - Exclude Search
+       * Finding child group if exists for selected group ids
+       */
+      $childGroupIds = CRM_Contact_BAO_Group::getChildGroupIds($this->_includeGroups);
+      if (count($childGroupIds) > 0) {
+        $this->_includeGroups = array_merge($this->_includeGroups, $childGroupIds);
+      }
+      //-- End - CRM-15049 - Bug Fix for Include - Exclude Search--//
```
Comment (jitendrapurohit) [inline CRM/Contact/Form/Search/Custom/Group.php]: Can you please remove this end comment line from the fix. Also I think a simple one line would be enough to make this understand of what this code is doing. Something like -

    //CRM-15049 - Include child group ids.
    <Fix for including child groups>

Comment (ahmedg1986) [inline CRM/Contact/Form/Search/Custom/Group.php]: This has been fixed now

**Review by jitendrapurohit — APPROVED** [review]

**Comment by twomice** [comment]
@colemanw or @kurund (I see you've commented on related tickets). I see @jitendrapurohit approved; is it possible to merge this or does it need more thorough documentation of review?

**Comment by samuelsov** [comment]
It works and it's an improvement but it's a fix only for Included groups, not for excluded groups.
I'd like this PR to be commited but as a next step we might want to have a helper for any custom search that want to add group filters.

**Comment by eileenmcnaughton** [comment]
Per previous comments & review- this does provide a partial fix and while I would prefer a unit test the length of time that has expired dictates an up or down approach to this & I agree with @samuelsov that it is better in than out.

Before
screenshot 2018-01-16 14 44 09


After
screenshot 2018-01-16 14 45 20

**Pull request merged**